test/jdk/jdk/internal/loader/URLClassPath/ClassnameCharTest.java

```diff
@@ -22,29 +22,29 @@
  */
 
 /* @test
- * @bug 4957669 5017871
+ * @bug 4957669 5017871 8358729
  * @summary cannot load class names containing some JSR 202 characters;
  *          plugin does not escape unicode character in http request
+ * @library /test/lib
  * @modules java.base/sun.net.www
  *          jdk.httpserver
  * @compile -XDignore.symbol.file=true ClassnameCharTest.java
  * @run main ClassnameCharTest
  */
 
 import java.io.*;
+import java.lang.classfile.ClassFile;
+import java.lang.constant.ClassDesc;
 import java.net.*;
-import java.security.AccessControlContext;
-import java.security.AccessController;
 import java.security.CodeSource;
-import java.security.PrivilegedActionException;
-import java.security.PrivilegedExceptionAction;
 import java.util.jar.*;
 import com.sun.net.httpserver.*;
+import jdk.test.lib.Utils;
 import sun.net.www.ParseUtil;
 
 public class ClassnameCharTest {
-    static String FNPrefix = System.getProperty("test.src", ".") + File.separator;
-    static File classesJar = new File(FNPrefix + "testclasses.jar");
+    private static final String JAR_PATH = Utils.TEST_CLASSES + Utils.FILE_SEPARATOR + "testclasses.jar";
```

Comment:
I think it would be better to use `java.nio.file.Path` which is like:

```java
private static final Path JAR_PATH = Path.of(".").resolve("testclasses.jar");
```
That way we don't have to reference the `Utils.TEST_CLASSES`. `Path.of(".")` will end up being the scratch directory of the test so jtreg can then retain this JAR file if the test fails for any reason.

Reply:
Thanks, this is cleaner.